turbo/snapshotsync/freezeblocks/beacon_block_reader.go

```diff
@@ -59,6 +59,9 @@ func (r *beaconSnapshotReader) ReadBlockBySlot(ctx context.Context, tx kv.Tx, sl
 	var buf []byte
 	if slot > r.sn.BlocksAvailable() {
 		data, err := r.beaconDB.GetBlock(ctx, tx, slot)
+		if data == nil {
+			return nil, err
```

Comment:
what if data == nil && err == nil ? should return an error ?

Reply:
i think i should do `if err != nil` first

Reply:
oh not this good actually, because if err = nil, it will just return nil, if not err. this logic is correct

Reply:
i am not sure if returning (nil, nil) is good ux.